codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/DocumentShapeDeserVisitor.java

```diff
@@ -411,7 +411,7 @@ protected final void generateDeserFunction(
             BiConsumer<GenerationContext, Shape> functionBody
     ) {
         SymbolProvider symbolProvider = context.getSymbolProvider();
-        GoWriter writer = context.getWriter();
+        GoWriter writer = context.getWriter().get();
```

Comment:
The ProtocolGenerator context is kind of a bit of a grab bag currently, in some instances a writer is to be expected, in other instances the delegator that is stored on the context is used and no writer is expected. This was just to make that contact a bit clearer that it may not be there. 

This was just something that I noticed when I was refactoring context to make it immutable after construction.